ui/components/institutional/qr-code-modal/qr-code-modal.js

```diff
@@ -0,0 +1,254 @@
+import React, { useState, useEffect, useContext, useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import QRCode from 'qrcode.react';
+import { v4 as uuid } from 'uuid';
+import { captureException } from '@sentry/browser';
+import {
+  Modal,
+  ModalOverlay,
+  Text,
+  Box,
+  ButtonLink,
+} from '../../component-library';
+import { ModalContent } from '../../component-library/modal-content/modal-content';
+import { ModalHeader } from '../../component-library/modal-header/modal-header';
+import {
+  TextAlign,
+  TextColor,
+  TextVariant,
+  FontWeight,
+} from '../../../helpers/constants/design-system';
+import { I18nContext } from '../../../contexts/i18n';
+import Spinner from '../../ui/spinner';
+import {
+  getChannelId,
+  getConnectionRequest,
+} from '../../../ducks/institutional/institutional';
+import { mmiActionsFactory } from '../../../store/institutional/institution-background';
+
+export default function QRCodeModal({
+  onClose,
+  custodianName = 'custodian',
+  custodianURL,
+  setQrConnectionRequest,
+}) {
+  const t = useContext(I18nContext);
+  const dispatch = useDispatch();
+  const channelId = useSelector(getChannelId);
+  const connectionRequest = useSelector(getConnectionRequest);
+  const [publicKey, setPublicKey] = useState(null);
+  const [privateKey, setPrivateKey] = useState(null);
+  const [decryptedMessage, setDecryptedMessage] = useState(null);
+  const [qrCodeValue, setQrCodeValue] = useState(null);
+  const [error, setError] = useState('');
+  const [traceId] = useState(uuid());
+  const mmiActions = mmiActionsFactory();
+
+  const handleClose = useCallback(async () => {
+    await dispatch(mmiActions.setConnectionRequest(null));
+    onClose();
+  }, [dispatch, mmiActions, onClose]);
+
+  const handleError = useCallback((message, e) => {
```

Comment:
Could extract this as a util function. wdyt?

Reply:
I think that in this case it doesn't provide any benefit as it's only being used inside this component and it's just for MMI context.